src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,89 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String DATA_SPLITTER = " ";
+    private static final String COMMA_SPLITTER = ",";
+    private static final int NAME_INDEX = 0;
+    private static final int COUNT_INDEX = 1;
+    private static final String SEPARATOR = System.lineSeparator();
+
+    public void getStatistic(String fromFileName, String toFileName) throws RuntimeException {
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            File file = new File(fromFileName);
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+            String input = bufferedReader.readLine();
+            while (input != null) {
+                stringBuilder.append(input).append(DATA_SPLITTER);
+                input = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file", e);
+        }
+        String strInFile = stringBuilder.toString();
+        String[] names = splitNames(strInFile);
+        String[] number = splitNumber(strInFile);
+        int supplyCounter = 0;
+        int buyCounter = 0;
+        File file1 = new File(toFileName);
+        BufferedWriter bufferedWriter = null;
+        try {
+            bufferedWriter = new BufferedWriter(new FileWriter(file1));
+            for (int i = 0;i < names.length;i++) {
+                if (names[i].equals("supply")) {
+                    supplyCounter += Integer.parseInt(number[i]);
+                } else {
+                    buyCounter += Integer.parseInt(number[i]);
+                }
+            }
+            bufferedWriter.write(writeInFile(supplyCounter,buyCounter));
+
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write in file",e);
+        } finally {
+            try {
+                if (bufferedWriter != null) {
+                    bufferedWriter.close();
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Can't close file",e);
+            }
+        }
```

Comment:
enormous big method. use this template to split it:
```suggestion
        String dataFromFile = readFile(fromFileName);
        String report = createReport(dataFromFile);
        writeReportToFile(report, toFileName);
```